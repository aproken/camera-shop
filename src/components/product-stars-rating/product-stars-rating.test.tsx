import { render, RenderResult } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import ProductStarsRating from './product-stars-rating';

describe('ProductStarsRating', () => {
  test('рендерит правильное количество звезд', () => {
    const rating = 3.5;
    const totalReview = 10;

    const { container }: RenderResult = render(<ProductStarsRating rating={rating} totalReview={totalReview} />);

    expect(container).toMatchSnapshot();
  });
});
