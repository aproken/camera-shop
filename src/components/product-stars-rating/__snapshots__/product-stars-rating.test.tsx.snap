// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductStarsRating рендерит правильное количество звезд 1`] = `
<div>
  <div
    class="rate product__rate"
  >
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-full-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-full-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-full-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-star"
      />
    </svg>
    <p
      class="visually-hidden"
    >
      Рейтинг: 
      3
    </p>
    <p
      class="rate__count"
    >
      <span
        class="visually-hidden"
      >
        Всего оценок:
      </span>
      10
    </p>
  </div>
</div>
`;
